lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/util/HashFactory.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.di.ipv.cri.drivingpermit.api.util;
+
+import uk.gov.di.ipv.cri.drivingpermit.api.domain.dva.request.DvaPayload;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+
+public class HashFactory {
+    private final Sha256MessageDigestFactory messageDigestFactory;
+
+    public HashFactory() {
+        this(new Sha256MessageDigestFactory());
+    }
+
+    public HashFactory(Sha256MessageDigestFactory messageDigestFactory) {
+        this.messageDigestFactory = messageDigestFactory;
+    }
+
+    public String getHash(DvaPayload request) throws NoSuchAlgorithmException {
+        String message =
+                request.getIssuerId()
+                        + request.getRequestId()
+                        + Objects.toString(request.getSurname(), "")
+                        + request.getForenames().stream()
+                                .filter(Objects::nonNull)
+                                .reduce("", String::concat)
+                        + Objects.toString(request.getDateOfBirth(), "")
+                        + request.getIssueDate()
+                        + request.getExpiryDate()
+                        + request.getDriverLicenceNumber()
+                        + Objects.toString(request.getPostcode(), "");
+        MessageDigest sha256;
+        try {
+            sha256 = messageDigestFactory.getInstance();
+        } catch (NoSuchAlgorithmException e) {
+            throw e;
```

Comment:
Can we make sure were logging this exception

Reply:
Logged.